controllers/medusa/medusarestorejob_controller.go

```diff
@@ -262,26 +263,36 @@ func (r *MedusaRestoreJobReconciler) podTemplateSpecUpdateComplete(ctx context.C
 }
 
 // prepareRestore prepares the MedusaRestoreMapping for the restore operation.
-// It uses the Medusa client to get the host map for the restore operation, using the Medusa standalone service to get the backup topology.
+// It uses the Medusa client to get the host map for the restore operation, using the first pod answering on the backup sidecar port.
 func (r *MedusaRestoreJobReconciler) prepareRestore(ctx context.Context, request *medusa.RestoreRequest) (*medusav1alpha1.MedusaRestoreMapping, error) {
-	medusaClient, err := r.ClientFactory.NewClient(ctx, medusaServiceUrl(
-		cassdcapi.CleanupForKubernetes(request.Datacenter.Spec.ClusterName),
-		request.Datacenter.SanitizedName(),
-		request.Datacenter.Namespace))
-	if err != nil {
-		return nil, err
-	}
-	restoreHostMap, err := medusa.GetHostMap(request.Datacenter, *request.RestoreJob, medusaClient, ctx)
+	pods, err := medusa.GetCassandraDatacenterPods(ctx, request.Datacenter, r, request.Log)
 	if err != nil {
+		request.Log.Error(err, "Failed to get datacenter pods")
 		return nil, err
 	}
 
-	medusaRestoreMapping, err := restoreHostMap.ToMedusaRestoreMapping()
-	if err != nil {
-		return nil, err
-	}
+	for _, pod := range pods {
+		addr := net.JoinHostPort(pod.Status.PodIP, fmt.Sprint(shared.BackupSidecarPort))
+		if medusaClient, err := r.ClientFactory.NewClient(ctx, addr); err != nil {
+			request.Log.Error(err, "Failed to create Medusa client", "address", addr)
+		} else {
+			if err != nil {
```

Comment:
Issue: This logic doesn't appear to make very much sense. You're checking for an error, then logging the error, then checking for the error again and returning it?? What am I missing here?

Reply:
Thought: this all really needs a refactor too, e.g. why is this logger sitting on the request struct not the reconciler struct?

Reply:
changed the logger and the method signature 👍